subprojects/core/src/main/java/org/gradle/initialization/DefaultGradleLauncher.java

```diff
@@ -141,6 +145,10 @@ private void doBuildStages(Stage upTo) {
             "Stage.Finished is not supported by doBuildStages."
         );
         try {
+            if (stage == null && gradle.isRootBuild()) {
+                buildOptionBuildOperationProgressEventsEmitter.emit(gradle.getStartParameter());
```

Comment:
Would it make sense to emit the file system watching build option at the place where it is observed in `FileSystemWatchingBuildActionRunner`? Same for the configuration cache parameter. That would at least ensure that the build option captured in build scans and the build option used in the build are the same.

The disadvantage would be that the operations are emitted from different places for the different build options.

Reply:
I can't recall the details now, but notifying when the value is observed is where I started and couldn't make it work due to some limitation. 

We can't certainly try again if you feel strongly about it.